### PR TITLE
[hsa-rocr] Remove debug_print in LibLoad

### PR DESCRIPTION
**Pull request by tpkessler**
This is a follow-up PR, see #112  and the more recent #116.

### CODE REVIEW

File: hsa-rocr/PKGBUILD [hunk 1]
```diff
@@ -15,15 +15,21 @@ makedepends=('cmake' 'libelf' "hsakmt-roct")
 provides=("rocr-runtime=$pkgver")
 replaces=('rocr-runtime')
 conflicts=('rocr-runtime')
-source=("$url/archive/rocm-$pkgver.tar.gz")
-sha256sums=('fa2d2d1f8a61d8a6952d377cf288d78c61776c3c2a666f163cafc3aa19ab0b61')
+source=("$url/archive/rocm-$pkgver.tar.gz"
+    'remove-warnings.patch')
```
Comment (ranisalt) [inline hsa-rocr/PKGBUILD:19]: Please align it to the " above

File: hsa-rocr/PKGBUILD [hunk 2]
```diff
@@ -15,15 +15,21 @@ makedepends=('cmake' 'libelf' "hsakmt-roct")
...
+sha256sums=('fa2d2d1f8a61d8a6952d377cf288d78c61776c3c2a666f163cafc3aa19ab0b61'
+    '008cd076e6d30215b3e645586ae999aaaadfd7606d0b1ab313daa78d791440c6')
```
Comment (ranisalt) [inline hsa-rocr/PKGBUILD:21]: Please align it to the ' above

File: hsa-rocr/PKGBUILD [hunk 3]
```diff
@@ -15,15 +15,21 @@ makedepends=('cmake' 'libelf' "hsakmt-roct")
...
 _dirname="$(basename "$url")-$(basename "${source[0]}" .tar.gz)"
 
+prepare() {
+    cd "$srcdir/$_dirname/src"
```
Comment (ranisalt) [inline hsa-rocr/PKGBUILD:25]: `makepkg` already runs in `$srcdir`, use `cd "$_dirname/src"`

File: hsa-rocr/PKGBUILD [hunk 4]
```diff
@@ -15,15 +15,21 @@ makedepends=('cmake' 'libelf' "hsakmt-roct")
...
+    patch -p0 < "$srcdir/remove-warnings.patch"
```
Comment (ranisalt) [inline hsa-rocr/PKGBUILD:26]: Other patches usually run `patch -Np1 -i <patch file>`, can you see if that works here too? I'd rather have the same behavior everywhere.

File: hsa-rocr/PKGBUILD [hunk 5]
```diff
@@ -15,15 +15,21 @@ makedepends=('cmake' 'libelf' "hsakmt-roct")
...
+}
+
```
Comment (ranisalt) [inline hsa-rocr/PKGBUILD:28]: This function is using 4 spaces while the rest of the file uses 2, please change this.

**Comment by tpkessler** [comment]
Hey @ranisalt! Thanks for your quick review. I'm glad you like it! :) I pushed the changes you requested.

**Comment by ranisalt** [comment]
`return *(LibHandle*)&ret;` is some of the worst code I've seen, clearly someone with a heavy C background simply didn't learn about `dynamic_cast` :laughing:

**Pull request merged**